code/modules/aquarium/fish.dm

```diff
@@ -0,0 +1,143 @@
+// Fish path used for autogenerated fish
+/obj/item/fish
+	name = "generic looking aquarium fish"
+	desc = "very bland"
+	w_class = WEIGHT_CLASS_TINY
+
+/// Automatically generates object of given base path from the behaviour type in loc
+/proc/generate_fish(loc,behaviour_type,base_path=/obj/item/fish)
+	var/datum/aquarium_behaviour/behaviour = behaviour_type
+	var/obj/item/fish = new /obj/item/fish(loc)
+	fish.name = initial(behaviour.name)
+	fish.icon = initial(behaviour.icon)
+	fish.icon_state = initial(behaviour.icon_state)
+	fish.desc = initial(behaviour.desc)
+	if(initial(behaviour.color))
+		fish.add_atom_colour(initial(behaviour.color), FIXED_COLOUR_PRIORITY)
+	if(ispath(behaviour_type,/datum/aquarium_behaviour/fish))
+		var/datum/aquarium_behaviour/fish/fish_behaviour = behaviour_type
+		var/fillet_type = initial(fish_behaviour.fillet_type)
+		if(fillet_type)
+			fish.AddElement(/datum/element/processable, TOOL_KNIFE, fillet_type, 1, 5)
+	fish.AddElement(/datum/element/deferred_aquarium_content, behaviour_type)
+	return fish
+
+/// Returns random fish, using random_case_rarity probabilities.
+/proc/random_fish_type(case_fish_only=TRUE,required_fluid)
+	var/static/probability_table
+	var/argkey = "fish_[required_fluid]_[case_fish_only]" //If this expands more extract bespoke element arg generation to some common helper.
+	if(!probability_table || !probability_table[argkey])
+		if(!probability_table)
+			probability_table = list()
+		var/chance_table = list()
+		for(var/_fish_behavior in subtypesof(/datum/aquarium_behaviour/fish))
+			var/datum/aquarium_behaviour/fish/fish_behavior = _fish_behavior
+			if(required_fluid && initial(fish_behavior.required_fluid_type) != required_fluid)
+				continue
+			if(initial(fish_behavior.availible_in_random_cases) || !case_fish_only)
+				chance_table[fish_behavior] = initial(fish_behavior.random_case_rarity)
+		probability_table[argkey] = chance_table
+	return pickweight(probability_table[argkey])
+
+// Actual fish definitions below - there's no specific paths, they are autogenerated from behaviours
+
+// Freshwater fish
+
+/datum/aquarium_behaviour/fish/goldfish
+	name = "goldfish"
+	desc = "not actually made of gold"
+	icon_state = "goldfish"
+	sprite_width = 8
+	sprite_height = 8
+
+/datum/aquarium_behaviour/fish/angelfish
+	name = "angelfish"
+	desc = "Omnivore fish originating from Amazon River."
+	icon_state = "angelfish"
+	dedicated_in_aquarium_icon_state = "bigfish"
+	sprite_height = 7
+	source_height = 7
+
+/datum/aquarium_behaviour/fish/guppy
+	name = "guppy"
+	desc = "One of the most popular aquarium fish."
```

Comment:
```suggestion

/datum/aquarium_behaviour/fish/guppy
	name = "guppy"
	desc = "Guppy is also known as rainbow fish because of the brightly colored body and fins."
```